### PR TITLE
ci-operator: add etcd

### PR DESCRIPTION
**Pull request by hexfusion**
Starting the process of adding etcd release configs

depends on
https://github.com/openshift/ocp-build-data/pull/40

### CODE REVIEW

File: ci-operator/config/openshift/etcd/OWNERS [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# DO NOT EDIT; this file is auto-generated using tools/populate-owners.
+# from https://github.com/openshift/etcd/blob/ea0cf681c7f0278eff72846a6bfcd017cdfc83bf/OWNERS
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
+
+approvers:
```
Comment (smarterclayton) [inline ci-operator/config/openshift/etcd/OWNERS]: Unfortunately this is not who can approve these changes.  @stevekuznetsov how do we exclude autogenerated owners / override?  etcd has an upstream, we have a fork, the fork needs tighter control

File: ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+base_images:
+  base:
+    cluster: https://api.ci.openshift.org
+    name: origin-v4.0
+    namespace: openshift
+    tag: base
+canonical_go_repository: github.com/openshift/etcd
```
Comment (smarterclayton) [inline ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml]: canonical would be `github.com/etcd-io/etcd`

File: ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+base_images:
```
Comment (smarterclayton) [inline ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml]: One problem with your branch names is that if you change anything about the branch your CI has to get recreated.  I would recommend creating `openshift-4.0` and tracking that here instead.

File: ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  tag: ''
+  tag_overrides: {}
+build_root:
+  image_stream_tag:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+tests:
+- as: unit
```
Comment (smarterclayton) [inline ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml]: you'll want to add an e2e-aws one here as well

Comment (hexfusion) [reply ci-operator/config/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10.yaml]: not sure I completely understand this usage let me know if I need something more custom.

File: ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.0__rhel.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  base:
+    name: '4.0'
+    namespace: ocp
+    tag: base
+build_root:
+  image_stream_tag:
+    name: builder
+    namespace: ocp
+    tag: golang-1.10
+canonical_go_repository: github.com/openshift/etcd
```
Comment (smarterclayton) [inline ci-operator/config/openshift/etcd/openshift-etcd-openshift-4.0__rhel.yaml]: etcd-io

File: ci-operator/job/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10-postsubmits.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  openshift/etcd:
+  - agent: kubernetes
+    branches:
+    - ^origin-4\.0-etcd-v3\.3\.10$
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/variant: rhel
+    name: branch-ci-openshift-etcd-origin-4.0-etcd-v3.3.10-rhel-images
```
Comment (smarterclayton) [inline ci-operator/job/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10-postsubmits.yaml]: You'll need to delete all the job/* files and let ci-operator-prowgen recreate them

Comment (hexfusion) [reply ci-operator/job/openshift/etcd/openshift-etcd-origin-4.0-etcd-v3.3.10-postsubmits.yaml]: ack

**Comment by smarterclayton** [comment]
Oops, we missed something.  You got RHEL variants, but not normal variants.

**Pull request merged**